docs/reference/arrow_support.rst

```diff
@@ -0,0 +1,88 @@
+.. _arrow-support:
+
+=============
+Arrow Support
+=============
+
+`Arrow <https://arrow.apache.org/>`__
+is an in-memory data exchange format that is the spiritual
+successor to the NumPy array interface. It provides for zero-copy
+access to columnar data, which in our case is ``Image`` data.
+
+The goal with Arrow is to provide native zero-copy interoperability
+with any Arrow provider or consumer in the Python ecosystem.
+
+.. warning:: Zero-copy does not mean zero allocation -- the internal
+  memory layout of Pillow images contains an allocation for row
+  pointers, so there is a non-zero, but significantly smaller than a
+  full-copy memory cost to reading an Arrow image.
+
+
+Data Formats
+============
+
+Pillow currently supports exporting Arrow images in all modes
+**except** for ``BGR;15``, ``BGR;16`` and ``BGR;24``. This is due to
+line-length packing in these modes making for non-continuous memory.
```

Comment:
Fyi, those modes are also deprecated - https://github.com/python-pillow/Pillow/pull/7978

Reply:
They exist, and I want to state _why_ they're not supported, in case we wind up with other similar image formats.